library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphTest.java

```diff
@@ -224,19 +396,122 @@ public void shouldAddAndGetEdge() {
         // Then
         final Edge edge = edges.next();
         assertFalse(edges.hasNext()); // there is only 1 vertex
-        assertEquals(VERTEX_1, ((List) edge.id()).get(0));
-        assertEquals(VERTEX_2, ((List) edge.id()).get(1));
-        assertEquals(CREATED_EDGE_GROUP, edge.label());
-        assertEquals(gafferPopInVertex, edge.inVertex());
-        assertEquals(gafferPopOutVertex, edge.outVertex());
-        assertEquals(1.5, (Double) edge.property(WEIGHT_PROPERTY).value(), 0);
+        assertThat(((List) edge.id()).get(0)).isEqualTo(VERTEX_1);
+        assertThat(((List) edge.id()).get(1)).isEqualTo(VERTEX_2);
+        assertThat(edge.label()).isEqualTo(CREATED_EDGE_GROUP);
+        assertThat(edge.inVertex()).isEqualTo(gafferPopInVertex);
+        assertThat(edge.outVertex()).isEqualTo(gafferPopOutVertex);
+        assertThat((Double) edge.property(WEIGHT_PROPERTY).value()).isEqualTo(1.5);
+    }
+
+    @Test
+    public void shouldGetEdgeInGroup() {
+        // Given
+        final Graph gafferGraph = getGafferGraph();
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
+        final GafferPopEdge edgeToAdd1 = new GafferPopEdge(CREATED_EDGE_GROUP, VERTEX_1, VERTEX_2, graph);
+        graph.addEdge(edgeToAdd1);
+
+        // When
+        final Iterator<GafferPopEdge> edges = graph.edges(VERTEX_1, Direction.OUT, CREATED_EDGE_GROUP);
+
+        // Then
+        final List<Edge> edgesList = new ArrayList<>();
+        while (edges.hasNext()) {
+            edgesList.add(edges.next());
+        }
+        assertThat(edgesList).contains(edgeToAdd1);
+    }
+
+    @Test
+    public void shouldGetEdgeInGroupWithViewWithEntity() {
+        // Given
+        final Graph gafferGraph = getGafferGraph();
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
+        final GafferPopEdge edgeToAdd1 = new GafferPopEdge(CREATED_EDGE_GROUP, VERTEX_1, VERTEX_2, graph);
+        graph.addEdge(edgeToAdd1);
+        final View view = new View.Builder()
+                .entity(CREATED_EDGE_GROUP)
```

Comment:
This is strange, you are asking for entities of group `CREATED_EDGE_GROUP` and it returns edge results?